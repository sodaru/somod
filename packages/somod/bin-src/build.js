/* eslint-disable */
const { join, dirname } = require("path");
const { mkdir, writeFile, chmod } = require("fs/promises");

const template = `#!/usr/bin/env node
/* eslint-disable */
require("cli-opentelemetry").tele(
  "somod",
  require("path").join(__dirname, "../dist/index.js"),
  "${process.env.GITHUB_REF_NAME || ""}",
  "${process.env.OTLP_URL || ""}",
  { "${process.env.OTLP_KEY_NAME || ""}": "${
  process.env.OTLP_KEY_VALUE || ""
}" },
  2 * 60 * 1000 // 2 minutes
);
`;

async function build() {
  const binFilePath = join(__dirname, "../bin/somod.js");
  await mkdir(dirname(binFilePath), { recursive: true });
  await writeFile(binFilePath, template);
  await chmod(binFilePath, 0755);
}

build();
