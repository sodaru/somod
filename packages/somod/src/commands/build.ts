import { CommonOptions, taskRunner, Command } from "nodejs-cli-runner";
import {
  buildParameters,
  buildServerlessTemplate,
  buildUiConfigYaml,
  buildUiPublic,
  compileTypeScript,
  deleteBuildDir,
  validatePageData,
  validatePageExports,
  file_configYaml,
  file_packageJson,
  file_parametersYaml,
  file_templateYaml,
  file_tsConfigSomodJson,
  findRootDir,
  isValidTsConfigSomodJson,
  key_somod,
  path_build,
  path_pages,
  path_public,
  path_serverless,
  path_ui,
  savePackageJson,
  updateSodaruModuleKeyInPackageJson,
  validatePackageJson,
  validateParametersWithSchema,
  validateServerlessTemplateWithSchema,
  validateUiConfigYaml,
  path_pagesData,
  loadPlugins,
  runPluginPrebuild,
  runPluginBuild,
  validateServerlessTemplate,
  initializeModuleHandler,
  validateUiConfigYamlWithSchema,
  loadNamespaces,
  path_functions,
  validateFunctionExports
} from "somod-lib";
import {
  addSOMODCommandTypeOptions,
  getSOMODCommandTypeOptions,
  SOMODCommandTypeOptions
} from "../utils/common";

type BuildOptions = CommonOptions & SOMODCommandTypeOptions;

export const BuildAction = async ({
  verbose,
  ...options
}: BuildOptions): Promise<void> => {
  const dir = findRootDir();

  const { ui, serverless } = getSOMODCommandTypeOptions(options);

  const plugins = await loadPlugins(dir);

  await taskRunner(
    `Initialize ModuleHandler`,
    initializeModuleHandler,
    verbose,
    dir,
    plugins.namespaceLoaders
  );

  await Promise.all([
    taskRunner(
      `Validate ${file_packageJson}`,
      validatePackageJson,
      verbose,
      dir
    ),
    taskRunner(
      `Validate ${file_tsConfigSomodJson}`,
      isValidTsConfigSomodJson,
      verbose,
      dir,
      ui ? { jsx: "react-jsx" } : {},
      [...(ui ? [path_ui] : []), ...(serverless ? [path_serverless] : [])]
    ),
    taskRunner(
      `Validate ${file_parametersYaml} with schema`,
      validateParametersWithSchema,
      verbose,
      dir
    )
  ]);

  if (ui) {
    await Promise.all([
      taskRunner(
        `Validate ${path_ui}/${file_configYaml} with schema`,
        validateUiConfigYamlWithSchema,
        verbose,
        dir
      ),
      taskRunner(
        `Validate ${path_ui}/${file_configYaml}`,
        validateUiConfigYaml,
        verbose,
        dir,
        plugins.uiKeywords
      ),
      taskRunner(
        `Validate exports in ${path_ui}/${path_pages}`,
        validatePageExports,
        verbose,
        dir
      ),
      taskRunner(
        `Validate exports in ${path_ui}/${path_pagesData}`,
        validatePageData,
        verbose,
        dir
      )
    ]);
  }

  if (serverless) {
    await taskRunner(
      `Validate ${path_serverless}/${file_templateYaml} with schema`,
      validateServerlessTemplateWithSchema,
      verbose,
      dir
    );
    await taskRunner(
      `Validate ${path_serverless}/${file_templateYaml}`,
      validateServerlessTemplate,
      verbose,
      dir,
      plugins.serverlessKeywords
    );
    await taskRunner(
      `Validate exports in ${path_serverless}/${path_functions}`,
      validateFunctionExports,
      verbose,
      dir
    );
  }

  await taskRunner(`Resolve Namespaces`, loadNamespaces, verbose);

  await Promise.all(
    plugins.prebuild.map(plugin =>
      taskRunner(
        `PreBuild plugin ${plugin.name}`,
        runPluginPrebuild,
        verbose,
        dir,
        plugin.plugin,
        {
          ui,
          serverless
        }
      )
    )
  );

  await taskRunner(
    `Delete ${path_build} directory`,
    deleteBuildDir,
    verbose,
    dir
  );
  await taskRunner(`Compile Typescript`, compileTypeScript, verbose, dir);

  if (ui) {
    await taskRunner(
      `Build ${path_ui}/${path_public}`,
      buildUiPublic,
      verbose,
      dir
    );

    await taskRunner(
      `Build ${path_ui}/${file_configYaml}`,
      buildUiConfigYaml,
      verbose,
      dir
    );
  }

  if (serverless) {
    await taskRunner(
      `Build ${path_serverless}/${file_templateYaml}`,
      buildServerlessTemplate,
      verbose,
      dir
    );
  }

  await taskRunner(
    `Build ${file_parametersYaml}`,
    buildParameters,
    verbose,
    dir
  );

  await taskRunner(
    `Set ${key_somod} version in ${file_packageJson}`,
    updateSodaruModuleKeyInPackageJson,
    verbose,
    dir
  );
  await taskRunner(`Save ${file_packageJson}`, savePackageJson, verbose, dir);

  await Promise.all(
    plugins.build.map(plugin =>
      taskRunner(
        `Build plugin ${plugin.name}`,
        runPluginBuild,
        verbose,
        dir,
        plugin.plugin,
        {
          ui,
          serverless
        }
      )
    )
  );
};

const buildCommand = new Command("build");

buildCommand.action(BuildAction);
addSOMODCommandTypeOptions(buildCommand);

export default buildCommand;
