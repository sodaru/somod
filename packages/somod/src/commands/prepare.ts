import { CommonOptions, taskRunner, Command } from "nodejs-cli-runner";
import {
  createPages,
  createPublicAssets,
  deletePagesAndPublicDir,
  file_dotenv,
  file_nextConfigJs,
  file_parametersJson,
  file_templateYaml,
  findRootDir,
  generateNextConfig,
  generateRootParameters,
  prepareSAMTemplate,
  path_pages,
  path_public,
  initializeModuleHandler,
  bundleFunctions,
  bundleFunctionLayers,
  loadLifeCycleHooks,
  appendNamespaceLoaders,
  runPrepareLifeCycleHook,
  runPreprepareLifeCycleHook
} from "somod-lib";
import {
  addSOMODCommandTypeOptions,
  getSOMODCommandTypeOptions,
  SOMODCommandTypeOptions
} from "../utils/common";

type PrepareOptions = CommonOptions & SOMODCommandTypeOptions;

export const PrepareAction = async ({
  verbose,
  ...options
}: PrepareOptions): Promise<void> => {
  const dir = findRootDir();

  const { ui, serverless } = getSOMODCommandTypeOptions(options);

  await taskRunner(
    `Initialize ModuleHandler`,
    initializeModuleHandler,
    { verbose, progressIndicator: true },
    dir
  );

  const moduleLifeCycleHooks = await loadLifeCycleHooks();

  try {
    await appendNamespaceLoaders(moduleLifeCycleHooks.namespaceLoaders);
  } catch (e) {
    // NOTE:  This is the case when called from deploy,
    // In deploy, build command is called before prepare, so the namespaces are already resolved in the build step.
    if (
      e.message !=
      "new namespaces can not be appended after namespace are resolved"
    ) {
      throw e;
    }
  }

  await Promise.all(
    moduleLifeCycleHooks.preprepare.map(plugin =>
      taskRunner(
        `Run PrePrepare Hook of ${plugin.name}`,
        runPreprepareLifeCycleHook,
        { verbose, progressIndicator: true },
        dir,
        plugin.lifeCycle,
        {
          ui,
          serverless
        }
      )
    )
  );

  await taskRunner(
    `Create/Update /${file_parametersJson}`,
    generateRootParameters,
    { verbose, progressIndicator: true },
    dir
  );

  if (ui) {
    await taskRunner(
      `Deleting /${path_pages} and /${path_public}`,
      deletePagesAndPublicDir,
      { verbose, progressIndicator: true },
      dir
    );
    await taskRunner(
      `Create /${path_pages}`,
      createPages,
      { verbose, progressIndicator: true },
      dir
    );
    await taskRunner(
      `Create /${path_public}`,
      createPublicAssets,
      { verbose, progressIndicator: true },
      dir
    );

    await taskRunner(
      `Gernerate /${file_nextConfigJs} and /${file_dotenv}`,
      generateNextConfig,
      { verbose, progressIndicator: true },
      dir,
      moduleLifeCycleHooks.uiKeywords
    );
  }
  if (serverless) {
    await taskRunner(
      `Bundle Serverless Functions`,
      bundleFunctions,
      { verbose, progressIndicator: true },
      dir
    );

    await taskRunner(
      `Bundle Serverless FunctionLayers`,
      bundleFunctionLayers,
      { verbose, progressIndicator: true },
      dir,
      verbose
    );

    await taskRunner(
      `Generate /${file_templateYaml}`,
      prepareSAMTemplate,
      { verbose, progressIndicator: true },
      dir,
      moduleLifeCycleHooks.serverlessKeywords
    );
  }

  await Promise.all(
    moduleLifeCycleHooks.prepare.map(plugin =>
      taskRunner(
        `Run Prepare Hook of ${plugin.name}`,
        runPrepareLifeCycleHook,
        { verbose, progressIndicator: true },
        dir,
        plugin.lifeCycle,
        {
          ui,
          serverless
        }
      )
    )
  );
};

const prepareCommand = new Command("prepare");

prepareCommand.action(PrepareAction);
addSOMODCommandTypeOptions(prepareCommand);

export default prepareCommand;
