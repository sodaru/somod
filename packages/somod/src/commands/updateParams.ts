import { CommonOptions, taskRunner, Command, Option } from "nodejs-cli-runner";
import {
  file_parametersJson,
  findRootDir,
  updateParametersFromSAM
} from "somod-lib";

type UpdateParamsOptions = CommonOptions & {
  stackName?: string;
};

export const UpdateParamsAction = async ({
  verbose,
  stackName
}: UpdateParamsOptions): Promise<void> => {
  const dir = findRootDir();

  await taskRunner(
    `Updating /${file_parametersJson}`,
    updateParametersFromSAM,
    { verbose, progressIndicator: true },
    dir,
    stackName
  );
};

const updateParamsCommand = new Command("update-params");

updateParamsCommand.action(UpdateParamsAction);

updateParamsCommand.addOption(
  new Option(
    "-s, --stack-name <name>",
    "Stack name to update the params from. Reads from samconfig.toml if omitted"
  )
);

export default updateParamsCommand;
