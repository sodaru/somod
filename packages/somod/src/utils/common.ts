import { Command } from "nodejs-cli-runner";

export type SOMODCommandTypeOptions = {
  ui: boolean;
  serverless: boolean;
};

export type DebugModeOptions = {
  debug: boolean;
};

export const addSOMODCommandTypeOptions = (command: Command) => {
  command.option("--ui", "only ui");
  command.option("--serverless", "only serverless");
};

export const addDebugOptions = (command: Command) => {
  command.option("-d, --debug", "Enable Debug mode");
};

export const getSOMODCommandTypeOptions = (
  options: SOMODCommandTypeOptions
): SOMODCommandTypeOptions => {
  const ui = !options.serverless || options.ui;
  const serverless = !options.ui || options.serverless;
  return { ui, serverless };
};
