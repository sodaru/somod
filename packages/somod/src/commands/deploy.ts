import { CommonOptions, taskRunner, Command, Option } from "nodejs-cli-runner";
import { file_templateYaml, findRootDir, samDeploy } from "somod-lib";
import { addDebugOptions, DebugModeOptions } from "../utils/common";
import { BuildAction } from "./build";
import { PrepareAction } from "./prepare";

type DeployOptions = CommonOptions &
  DebugModeOptions & {
    guided: boolean;
  };

export const DeployAction = async ({
  verbose,
  guided,
  debug
}: DeployOptions): Promise<void> => {
  const dir = findRootDir();

  await BuildAction({ verbose, ui: false, serverless: true, debug });

  await PrepareAction({ verbose, ui: false, serverless: true, debug });

  await taskRunner(
    `Deploying ${file_templateYaml}`,
    samDeploy,
    { verbose, progressIndicator: false },
    dir,
    verbose,
    guided
  );
};

const deployCommand = new Command("deploy");

deployCommand.action(DeployAction);

deployCommand.addOption(
  new Option(
    "-g, --guided",
    "guided will assist in configuring backend parameters in apply state"
  )
);
addDebugOptions(deployCommand);

export default deployCommand;
