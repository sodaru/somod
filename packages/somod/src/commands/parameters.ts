import { CommonOptions, taskRunner, Command, Option } from "nodejs-cli-runner";
import {
  file_parametersJson,
  findRootDir,
  initializeContext,
  updateParametersFromSAM,
  validateParameterValues
} from "somod-lib";
import { addDebugOptions, DebugModeOptions } from "../utils/common";

type UpdateParamsOptions = CommonOptions &
  DebugModeOptions & {
    stackName?: string;
  };

type ValidateParamsOptions = CommonOptions & DebugModeOptions;

export const UpdateParamsAction = async ({
  verbose,
  debug,
  stackName
}: UpdateParamsOptions): Promise<void> => {
  const dir = findRootDir();

  const context = await taskRunner(
    `Initialize Context`,
    initializeContext,
    {
      verbose,
      progressIndicator: true
    },
    dir,
    false,
    true,
    debug
  );

  await taskRunner(
    `Updating /${file_parametersJson}`,
    updateParametersFromSAM,
    { verbose, progressIndicator: true },
    context,
    stackName
  );
};

const updateParamsCommand = new Command("update");

updateParamsCommand.action(UpdateParamsAction);

updateParamsCommand.addOption(
  new Option(
    "-s, --stack-name <name>",
    "Stack name to update the params from. Reads from samconfig.toml if omitted"
  )
);
addDebugOptions(updateParamsCommand);

export const ValidateParamsAction = async ({
  verbose,
  debug
}: ValidateParamsOptions): Promise<void> => {
  const dir = findRootDir();

  const context = await taskRunner(
    `Initialize Context`,
    initializeContext,
    {
      verbose,
      progressIndicator: true
    },
    dir,
    false,
    false,
    debug
  );

  await taskRunner(
    `Validating /${file_parametersJson}`,
    validateParameterValues,
    { verbose, progressIndicator: true },
    context
  );
};

const validateParamsCommand = new Command("validate");

validateParamsCommand.action(ValidateParamsAction);
addDebugOptions(validateParamsCommand);

const parametersCommand = new Command("parameters");
parametersCommand.addCommand(updateParamsCommand);
parametersCommand.addCommand(validateParamsCommand);

export default parametersCommand;
