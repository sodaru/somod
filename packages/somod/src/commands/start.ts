import { CommonOptions, taskRunner, Command, Option } from "nodejs-cli-runner";
import {
  findRootDir,
  nextCommand,
  path_pages,
  path_pagesData,
  path_public,
  path_ui,
  watchRootModulePages,
  watchRootModulePagesData,
  watchRootModulePublicAssets
} from "somod-lib";
import { addDebugOptions, DebugModeOptions } from "../utils/common";
import { BuildAction } from "./build";
import { PrepareAction } from "./prepare";

type StartOptions = CommonOptions &
  DebugModeOptions & {
    dev: boolean;
  };

export const StartAction = async ({
  verbose,
  debug,
  dev
}: StartOptions): Promise<void> => {
  const dir = findRootDir();

  await BuildAction({ verbose, ui: true, serverless: false, debug });

  await PrepareAction({ verbose, ui: true, serverless: false, debug });

  if (dev) {
    await taskRunner(
      `Watch ${path_ui}/${path_pages}`,
      watchRootModulePages,
      { verbose, progressIndicator: true },
      dir
    );
    await taskRunner(
      `Watch ${path_ui}/${path_pagesData}`,
      watchRootModulePagesData,
      { verbose, progressIndicator: true },
      dir
    );
    await taskRunner(
      `Watch ${path_ui}/${path_public}`,
      watchRootModulePublicAssets,
      { verbose, progressIndicator: true },
      dir
    );

    await taskRunner(
      `Start NextJs Development server`,
      nextCommand,
      { verbose, progressIndicator: false },
      dir,
      ["dev"]
    );
  } else {
    await taskRunner(
      `Build NextJS Project`,
      nextCommand,
      { verbose, progressIndicator: false },
      dir,
      ["build"]
    );
    await taskRunner(
      `Start NextJS Server`,
      nextCommand,
      { verbose, progressIndicator: false },
      dir,
      ["start"]
    );
  }
};

const startCommand = new Command("start");

startCommand.action(StartAction);

startCommand.addOption(
  new Option("-d, --dev", "Start a dev server in watch mode")
);
addDebugOptions(startCommand);
export default startCommand;
