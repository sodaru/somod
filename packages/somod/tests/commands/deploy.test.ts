import deployCommand from "../../src/commands/deploy";
import { BuildAction } from "../../src/commands/build";
import { PrepareAction } from "../../src/commands/prepare";
import { IContext } from "somod-types";
import * as lib from "somod-lib";
import { mockedFunction } from "../utils";
import { addCommonOptions } from "nodejs-cli-runner";

jest.mock("somod-lib", () => {
  const original = jest.requireActual("somod-lib");
  const mocked = {};
  for (const key in original) {
    mocked[key] =
      typeof original[key] === "function" ? jest.fn() : original[key];
  }
  return {
    __esModule: true,
    ...mocked
  };
});

jest.mock("../../src/commands/build", () => {
  return {
    __esModule: true,
    BuildAction: jest.fn()
  };
});

jest.mock("../../src/commands/prepare", () => {
  return {
    __esModule: true,
    PrepareAction: jest.fn()
  };
});

describe("Test command deploy", () => {
  let output: unknown[][] = [];

  // eslint-disable-next-line no-console
  const originalConsoleLog = console.log;

  const functionsToBeMocked = ["samDeploy"];

  const mock = (input: Record<string, unknown>) => {
    for (const key of functionsToBeMocked) {
      // eslint-disable-next-line import/namespace
      mockedFunction(lib[key]).mockImplementation(async (...args) => {
        output.push([key, ...args]);
        return input[key];
      });
    }
    mockedFunction(lib.findRootDir).mockImplementation((...args) => {
      output.push(["findRootDir", ...args]);
      return "/root/dir";
    });
    mockedFunction(BuildAction).mockImplementation(async (...args) => {
      output.push(["BuildAction", ...args]);
    });
    mockedFunction(PrepareAction).mockImplementation(async (...args) => {
      output.push(["PrepareAction", ...args]);
    });
  };

  beforeAll(() => {
    addCommonOptions(deployCommand);
  });

  beforeEach(() => {
    deployCommand.setOptionValue("verbose", undefined);
    deployCommand.setOptionValue("debug", undefined);
    deployCommand.setOptionValue("guided", undefined);

    // eslint-disable-next-line no-console
    console.log = originalConsoleLog;
    for (const key of functionsToBeMocked) {
      // eslint-disable-next-line import/namespace
      mockedFunction(lib[key]).mockReset();
      mockedFunction(BuildAction).mockReset();
      mockedFunction(PrepareAction).mockReset();
    }
    output = [];

    // eslint-disable-next-line no-console
    console.log = jest.fn();
  });

  test("without args", async () => {
    mock({});
    await deployCommand.parseAsync([], {
      from: "user"
    });
    expect(output).toMatchSnapshot();
  });

  test("with verbose", async () => {
    mock({});
    await deployCommand.parseAsync(["-v"], {
      from: "user"
    });
    expect(output).toMatchSnapshot();
  });

  test("with debug mode", async () => {
    mock({});
    await deployCommand.parseAsync(["-d"], {
      from: "user"
    });
    expect(output).toMatchSnapshot();
  });

  test("with guided", async () => {
    mock({});
    await deployCommand.parseAsync(["-g"], {
      from: "user"
    });
    expect(output).toMatchSnapshot();
  });

  test("with all options", async () => {
    mock({
      initializeContext: {
        dir: "/root/dir"
      } as IContext
    });
    await deployCommand.parseAsync(["-v", "-d", "-g"], {
      from: "user"
    });
    expect(output).toMatchSnapshot();
  });
});
