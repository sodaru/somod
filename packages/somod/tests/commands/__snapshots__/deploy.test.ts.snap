// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test command deploy with all options 1`] = `
Array [
  Array [
    "findRootDir",
  ],
  Array [
    "BuildAction",
    Object {
      "debug": true,
      "serverless": true,
      "ui": false,
      "verbose": true,
    },
  ],
  Array [
    "PrepareAction",
    Object {
      "debug": true,
      "serverless": true,
      "ui": false,
      "verbose": true,
    },
  ],
  Array [
    "samDeploy",
    "/root/dir",
    true,
    true,
  ],
]
`;

exports[`Test command deploy with debug mode 1`] = `
Array [
  Array [
    "findRootDir",
  ],
  Array [
    "BuildAction",
    Object {
      "debug": true,
      "serverless": true,
      "ui": false,
      "verbose": undefined,
    },
  ],
  Array [
    "PrepareAction",
    Object {
      "debug": true,
      "serverless": true,
      "ui": false,
      "verbose": undefined,
    },
  ],
  Array [
    "samDeploy",
    "/root/dir",
    undefined,
    undefined,
  ],
]
`;

exports[`Test command deploy with guided 1`] = `
Array [
  Array [
    "findRootDir",
  ],
  Array [
    "BuildAction",
    Object {
      "debug": undefined,
      "serverless": true,
      "ui": false,
      "verbose": undefined,
    },
  ],
  Array [
    "PrepareAction",
    Object {
      "debug": undefined,
      "serverless": true,
      "ui": false,
      "verbose": undefined,
    },
  ],
  Array [
    "samDeploy",
    "/root/dir",
    undefined,
    true,
  ],
]
`;

exports[`Test command deploy with verbose 1`] = `
Array [
  Array [
    "findRootDir",
  ],
  Array [
    "BuildAction",
    Object {
      "debug": undefined,
      "serverless": true,
      "ui": false,
      "verbose": true,
    },
  ],
  Array [
    "PrepareAction",
    Object {
      "debug": undefined,
      "serverless": true,
      "ui": false,
      "verbose": true,
    },
  ],
  Array [
    "samDeploy",
    "/root/dir",
    true,
    undefined,
  ],
]
`;

exports[`Test command deploy without args 1`] = `
Array [
  Array [
    "findRootDir",
  ],
  Array [
    "BuildAction",
    Object {
      "debug": undefined,
      "serverless": true,
      "ui": false,
      "verbose": undefined,
    },
  ],
  Array [
    "PrepareAction",
    Object {
      "debug": undefined,
      "serverless": true,
      "ui": false,
      "verbose": undefined,
    },
  ],
  Array [
    "samDeploy",
    "/root/dir",
    undefined,
    undefined,
  ],
]
`;
